bukkit/src/main/java/org/wensheng/juicyraspberrypie/RemoteSession.java

```diff
@@ -429,11 +448,73 @@ private void handleEntityCommand(String c, String[] args, boolean entityIsPlayer
                 }
                 Location loc = entity.getLocation();
                 entity.teleport(parseLocation(x, y, z, loc.getPitch(), loc.getYaw()));
+                send("ok");
+                break;
+            }
+            case "enableControl": {
+                if (entity instanceof Mob mob) {
+                    if(!mob.hasAI()) {
+                        mob.getPersistentDataContainer().set(new NamespacedKey(plugin, "noAI"), PersistentDataType.BOOLEAN, true);
+                        mob.setAI(true);
+                    }
+                    final MobGoals mobGoals = Bukkit.getMobGoals();
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noMoveAI", GoalType.MOVE));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noTargetAI", GoalType.TARGET));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noJumpAI", GoalType.JUMP));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noLookAI", GoalType.LOOK));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noUnknownBehaviorAI", GoalType.UNKNOWN_BEHAVIOR));

```

Comment:
I would extract a List for the `noMoveAI` ... `noUnknownBehaviorAI` Strings and use a loop to ensure that the `disableControl` branch will undo the exact same ones.